Assignment3/PointAndGrid.py

```diff
@@ -0,0 +1,260 @@
+class Point:
+
+    def __init__(self, grid = None, coordinates = None, value = None):
+        """
+        For Point initialization with optional grid and coordinates.
+        :param grid: Grid
+        :param coordinates: Tuple(int, int)
+        :return: None
+        """
+        self.grid = grid
+        self.coordinates = coordinates
+        self.value = value
+        if grid and coordinates:
+            v = grid.getElement(coordinates)
+            if value and value != v:
+                raise AttributeError
+            else: self.value = v
+
+    def setGrid(self, gr):

```

Comment:
It's absolutely not necessary. By default I would entirely omit the `setGrid` method and just do:

```
point.grid = gre
```

If anything, the decorator should probably be used sparingly because it can mislead readers into thinking the attribute is directly read/written while in reality custom logic is being run. That is to say, if this setter also verifyied the type of `gr` before setting it I would not comment on the existence of the explicit setter at and would be satisfied that it is required. I would not suggest that the setter could be hidden by the decorator.

But if the setter just sets, then it would be more Pythonic to not have setter (and getter) functions but rather read/write the attributes directly.